website/docs/guides/offline.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Offline mode ⚡️
+sidebar_position: 51
+---
+
+:::caution
+Offline mode is an experimental feature.
+
+Please bear in mind that non-ideal user experience should be expected.
+If you encounter any bugs or have feedback to share, make sure to reach out to the maintenance team
+on [GitHub](https://github.com/VirtusLab/scala-cli).
+:::
+
+The offline mode for Scala CLI has been introduced to be used in two situations:
+- you want to have more control over the artifacts being downloaded
+- your development environment has restricted access to the Internet or certain web domains
+
+In this mode Scala CLI will only use local artifacts cached by coursier. Any attempts to download new artifacts will fail if they are not recoverable.
+This applies to everything that Scala CLI normally manages behind the scenes:
+- Scala language and compiler artifacts
+- JVM artifacts
+- Bloop artifacts
+- dependency artifacts
+
+## How to use the offline mode
+
+To enable offline mode pass the `--offline` flag to Scala CLI, e.g.:
+
+```bash ignore
+scala-cli run Main.scala --offline
+```
+
+It is also possible to use the `COURSIER_MODE` environment variable or `coursier.mode` java property.
+```bash ignore
+export COURSIER_MODE=offline
+```
+or
+```bash ignore
+scala-cli -Dcoursier.mode=offline run Main.scala 
+```
+
+## Changes in behaviour
+
+### Scala artifacts
+In offline mode Scala CLI will not perform any validation of the Scala version specified in the project.
+
+### JVM artifacts
+System JVM will be used or a fetch from local cache performed.
+If a different JVM version than the system one is required, it is best to export it to the `JAVA_HOME` environment variable.
+It is important to know, that currently if a version is specified with `--jvm` or `using jvm` Scala CLI will ignore the system JVM and try to fetch via coursier.
+
+To start the Bloop server a JVM with version above 17 is required, if it can't be found compilation will fall back to using `scalac` instead.
+
+### Bloop artifacts
+If no artifacts for Bloop are available compilation falls back to using `scalac` instead.
+
+### Dependency artifacts
+Any attempt to download a dependency will fail, so it is required to have all the dependencies cached locally before compiling.
+
+Dependencies that reside in local repositories like `~/.ivy2/local` will be resolved as usual.
+
+## Setting up the environment
+
+The easiest way to set up the environment is to use [Coursier](https://get-coursier.io).
```

Comment:
an example on how to copy deps to the coursier cache manually could be useful for cases where someone has to set up an isolated environment